src/Form/Tag.elm

```diff
@@ -0,0 +1,188 @@
+module Form.Tag exposing
+    ( FormType(..)
+    , Model
+    , Msg
+    , ParentMsg(..)
+    , initAddFormModel
+    , initEditFormModel
+    , update
+    , view
+    )
+
+import Data.Tag as Tag
+import Dict exposing (Dict)
+import Html exposing (Html)
+import Html.Attributes as Attributes
+import Ui.Button as Button
+import Ui.Form as Form
+import Ui.Input as Input
+import Util.Validate as Validate
+
+
+type Model
+    = Model ModelData
+
+
+type alias ModelData =
+    { errors : Errors
+    , tag : Tag.Model
+    , originalTag : Maybe Tag.Model
+    }
```

Comment:
Like here I think that using `Model` everywhere shows how it starts to be harder to read. You have `Model` as an opaque type for `ModelData` where you have `Tag.Model`... It feels little bit to unnecessary complicated to me.